### DISCUSSION TITLE
TP7 - Duda con la Parte A - Concatenación dinámica

### DISCUSSION THREAD

**Discussion by sanvaltos** [INGCOM-UNRN-P1/cursada-2023#198] [Preguntas y respuestas] [answered]
Buenas tardes, tengo una consulta con el inciso A del ejercicio 3 del TP7. 

Mi duda es con el prototipo que nos dan en el enunciado  ¿no estarían faltando las capacidades de las cadenas?

![image](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/141690405/457a9a08-1aa9-4ff1-9493-796c90a62bb6)
 
Y otra duda ¿el resultado de la concatenación debe guardarse en una nueva cadena o en alguna de las dos pasadas como argumentos?

**Comment by TeoInMaghine** [comment, marked as answer]
Podés calcular el largo seguro de las dos cadenas, y el resultado va en otra cadena que retornás (justamente por eso es `char* concatenaCadenas`).

**Comment by martinvilu** [comment]
Si usas memoria dinámica, crea una cadena nueva con el largo tradicional de ambas sumada.

Hay _menos_ riesgo de desbordamientos, ya que el bloque de memoria va a ser del tamaño correcto.

Siempre te podés encontrar con una cadena que no tiene el `\0` y termines copiando de más.

Con esto en mente, es correcto agregar las capacidades.
